test/core/miner/PreCommitBatcherTest.cpp

```diff
@@ -0,0 +1,242 @@
+   protected:
+    virtual void SetUp() {
+      seal_proof_type_ = RegisteredSealProof::kStackedDrg2KiBV1;
+      api_ = std::make_shared<FullNodeApi>();
+      sch_ = std::make_shared<SchedulerMock>();
+      miner_id = 42;
+      miner_address_ = Address::makeFromId(miner_id);
+      current_time_ = toTicks(std::chrono::duration_cast<std::chrono::seconds>(
+          std::chrono::system_clock::now().time_since_epoch()));
+
+      api::BlockHeader block;
+      block.height = 2;
+
+      auto tipset = std::make_shared<Tipset>(
+          TipsetKey(), std::vector<api::BlockHeader>({block}));
+
+      api_->ChainHead = [=]() -> outcome::result<TipsetCPtr> {
+        return outcome::success(tipset);
+      };
+
+      api_->StateMinerInfo =
+          [=](const Address &address,
+              const TipsetKey &tipset_key) -> outcome::result<MinerInfo> {
+        if (address == miner_address_) {
+          MinerInfo info;
+          info.seal_proof_type = seal_proof_type_;
+          return info;
+        }
+        return ERROR_TEXT("Error");
+      };
+
+      EXPECT_CALL(*sch_, now()).WillOnce(testing::Return(current_time_));
+      auto result = PreCommitBatcherImpl::makeBatcher(
+          toTicks(std::chrono::seconds(60)), api_, sch_, miner_address_);
+      batcher_ = result.value();
+      ASSERT_FALSE(result.has_error());
+    }
+
+    std::shared_ptr<FullNodeApi> api_;
+    std::shared_ptr<SchedulerMock> sch_;
+    std::shared_ptr<PreCommitBatcherImpl> batcher_;
+    Address miner_address_;
+    uint64_t miner_id;
+    RegisteredSealProof seal_proof_type_;
+    Ticks current_time_;
+    primitives::TokenAmount mutualDeposit = 0;
+  };
+
+  TEST_F(PreCommitBatcherTest, BatcherWrite) {
+    SectorInfo si = SectorInfo();
+    api::SectorPreCommitInfo precInf;
+    primitives::TokenAmount deposit = 10;
+    ASSERT_FALSE(batcher_->addPreCommit(si, deposit, precInf).has_error());
+  };
+
+  /**
+   * CallbackSend test is checking that after the scheduled time for a Precommit
+   * collecting, all the stored batcher's data will be published in a
+   * messagePool
+   */
+  TEST_F(PreCommitBatcherTest, CallbackSend) {
+    mutualDeposit = 0;
+    bool isCall = false;
+
+    api::BlockHeader block;
+    block.height = 2;
+
+    auto tipset = std::make_shared<Tipset>(
+        TipsetKey(), std::vector<api::BlockHeader>({block}));
+
+    api_->ChainHead = [&]() -> outcome::result<TipsetCPtr> {
+      return outcome::success(tipset);
+    };
+
+    api_->MpoolPushMessage = [&](const UnsignedMessage &msg,
+                                 const boost::optional<api::MessageSendSpec> &)
+        -> outcome::result<SignedMessage> {
+      if (msg.method == vm::actor::builtin::v0::miner::PreCommitSector::Number
+          && msg.value == mutualDeposit) {
+        isCall = true;
+        return SignedMessage{.message = msg, .signature = BlsSignature()};
+      }
+
+      return ERROR_TEXT("ERROR");
+    };
```

Comment:
API calls are duplicated across several tests. May I suggest to move them to `SetUp`?

Reply:
fixed